node/src/database/rocksdb.rs

```diff
@@ -7,130 +7,350 @@
+    OptimisticTransactionDB, OptimisticTransactionOptions, Options,
+    ReadOptions, SnapshotWithThreadMode, WriteOptions,
+};
+use std::{marker::PhantomData, path::Path, sync::Arc};
+
 enum TxType {
     ReadWrite,
     ReadOnly,
 }
 
-pub struct Backend {}
+const CF_LEDGER: &str = "cf_ledger";
+const CF_CANDIDATES: &str = "cf_candidates";
+
+pub struct Backend {
+    rocksdb: Arc<OptimisticTransactionDB>,
+}
 
 impl Backend {
-    fn begin_tx(&mut self, read_only: TxType) -> Transaction {
-        // TODO: rusk/issues/806 This should be addressed with another issue
-        // about integrating RocksDB
-        Transaction {}
+    fn begin_txn(
+        &self,
+        access_type: TxType,
+    ) -> Transaction<'_, OptimisticTransactionDB> {
+        // Create a new RocksDB transaction
+        let write_options = WriteOptions::default();
+        let mut txn_options = OptimisticTransactionOptions::default();
+
+        let inner = self.rocksdb.transaction_opt(&write_options, &txn_options);
+
+        // Borrow column families
+        let ledger_cf = self
+            .rocksdb
+            .cf_handle(CF_LEDGER)
+            .expect("ledger column family must exist");
+
+        let candidates_cf = self
+            .rocksdb
+            .cf_handle(CF_CANDIDATES)
+            .expect("candidates column family must exist");
+
+        let snapshot = self.rocksdb.snapshot();
+
+        Transaction::<'_, OptimisticTransactionDB> {
+            inner,
+            access_type,
+            candidates_cf,
+            ledger_cf,
+            snapshot,
+        }
     }
 }
 
 impl DB for Backend {
-    type T = Transaction;
+    type T<'a> = Transaction<'a, OptimisticTransactionDB>;
+
+    fn create_or_open(path: String) -> Self {
+        let mut opts = Options::default();
+        opts.create_if_missing(true);
+        opts.create_missing_column_families(true);
+        opts.set_level_compaction_dynamic_level_bytes(true);
+
+        Self {
+            rocksdb: Arc::new(
+                rocksdb_lib::OptimisticTransactionDB::open_cf(
+                    &opts,
+                    Path::new(&path),
+                    [CF_LEDGER, CF_CANDIDATES],
+                )
+                .expect("should be a valid database"),
+            ),
+        }
+    }
 
-    fn view<F>(&mut self, f: F) -> Result<()>
+    fn view<F>(&self, f: F) -> Result<()>
     where
-        F: FnOnce(&Transaction) -> Result<()>,
+        F: for<'a> FnOnce(Self::T<'a>) -> Result<()>,
     {
-        // Create read-only transaction
-        let mut tx = self.begin_tx(TxType::ReadOnly);
+        // Create a new read-only transaction
+        let tx = self.begin_txn(TxType::ReadOnly);
 
-        // If f returns err, no commit will be applied into backend
-        // storage
-        f(&tx)?;
-
-        // Release tx resources
-        tx.close();
+        // Execute all read-only transactions in isolation
+        f(tx)?;
```

Comment:
I'm against calling it simply `f`. It's not so easy to understand what it is

Reply:
There is a comment above it to explain its idea. Anyway, I'll rename it to `execute`, if that makes it clearer.